docs/sources/flow/reference/components/loki.process.md

```diff
@@ -530,6 +524,182 @@ Then, the second stage adds `user="John Doe"` to the label set of the log
+
+The fourth stage uses the LogQL selector to only execute on lines where
+`applbl="qux"`; that means it won't match on any of our input, and the nested
+json stage does not run.
+
+The fifth stage drops any entries from on lines where `applbl` is set to
+'bar' and the line contents that matches the regex `.*noisy error.*`. It also
+increments the `loki_process_dropped_lines_total` metric with a label
+reason="discard_noisy_errors".
+
+The final output stage changes the contents of the log line to be the value of
+`msg` from the extracted map. In this case, the first log entry's content is
+is changed to `app1 log line`.
+
+
+### drop block
+
+The `drop` inner block configures a filtering stage that drops log entries
+based on several options. If multiple options are provided, they're treated
+as AND clauses and must _all_ be true for the log entry to be dropped.
+To drop entries with an OR clause, specify multiple `drop` blocks in sequence.
+
+The following arguments are supported:
+
+Name                  | Type       | Description                                           | Default   | Required
+--------------------- | ---------- | ----------------------------------------------------- | --------- | --------
+`source`              | `string`   | Name from extracted data to parse. If empty or not defined, it uses the log message.    | `""` | no
+`expression`          | `string`   | A valid RE2 regular expression. | `""` | no
+`value`               | `string`   | If both `source` and `value` are specified, the stage drops lines where `value` exactly matches the source content. | `""` | no
+`older_than`          | `duration` | If specified, the stage drops lines whose timestamp is older than the current time minus this duration. | `""` | no
+`longer_than`         | `string`   | If specified, the stage drops lines whose size exceeds the configured value. | `""` | no
+`drop_counter_reason` | `string`   | A custom reason to report for dropped lines. | `"drop_stage"` | no
+
+The `expression` field needs to be a RE2 regex string. If `source` is empty or
+not provided, the regex attempts to match the log line itself. If source is
+provided, the regex attempts to match the corresponding value from the
+extracted map.
+
+On the other hand, the `value` field can only work with values from the
+extracted map, and must be specified together with `source`. Entries are
+dropped when there is an exact match between the two.
+
+Whenever an entry is dropped, the metric `loki_process_dropped_lines_total`
+is incremented. By default, the reason label is `drop_stage`, but a custom
+one can be provided by using the `drop_counter_reason` argument.
+
+The following stage drops log entries that contain the word `debug` _and_ are
+longer than 1KB.
+
+```
+stage {
+	drop {
+		expression  = ".*debug.*"
+		longer_than = "1KB"
+	}
+}
+```
+
+The following stages drop entries that are 24h or older, are longer than
+8KB, _or_ the extracted value of 'app' is equal to foo.
```

Comment:
I think you should say why there's an OR here -- because there are multiple `drop` blocks.